### ISSUE TITLE
TessBaseAPI stop() sometimes causes SIGSEGV

### ISSUE THREAD

**Issue by rmtheis** [rmtheis/tess-two#97]

**Comment by panzerfahrer** [comment]
I get a crash every time `stop()` is called.

Stacktrace:
```
28285-28473/com.example A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x14 in tid 28473 (AsyncTask #1)
196-196/? I/SELinux: SELinux: Loaded file_contexts contexts from /file_contexts.
196-196/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
196-196/? A/DEBUG: Build fingerprint: 'google/hammerhead/hammerhead:6.0.1/MMB29S/2489379:user/release-keys'
196-196/? A/DEBUG: Revision: '0'
196-196/? A/DEBUG: ABI: 'arm'
196-196/? A/DEBUG: pid: 28285, tid: 28473, name: AsyncTask #1  >>> com.example <<<
196-196/? A/DEBUG: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x14
196-196/? A/DEBUG:     r0 98d3f114  r1 00000014  r2 00000001  r3 00008001
196-196/? A/DEBUG:     r4 98d3f1b8  r5 98d3f114  r6 00000000  r7 00000001
196-196/? A/DEBUG:     r8 00000001  r9 00000414  sl 9768c008  fp 9768c41c
196-196/? A/DEBUG:     ip 000004b1  sp 98d3f108  lr a0d05d2b  pc a0cc52c8  cpsr 08070030
196-196/? A/DEBUG: backtrace:
196-196/? A/DEBUG:     #00 pc 000782c8  /data/app/com.example-1/lib/arm/libtess.so (ELIST_ITERATOR::set_to_list(ELIST*)+1)
196-196/? A/DEBUG:     #01 pc 000b8d27  /data/app/com.example-1/lib/arm/libtess.so (WERD::true_bounding_box() const+26)
196-196/? A/DEBUG:     #02 pc 000b8d69  /data/app/com.example-1/lib/arm/libtess.so (WERD::restricted_bounding_box(bool, bool) const+16)
196-196/? A/DEBUG:     #03 pc 000b8dd9  /data/app/com.example-1/lib/arm/libtess.so (WERD::bounding_box() const+8)
196-196/? A/DEBUG:     #04 pc 00084aab  /data/app/com.example-1/lib/arm/libtess.so (tesseract::Tesseract::RecogAllWordsPassN(int, ETEXT_DESC*, PAGE_RES_IT*, GenericVector<tesseract::WordData>*)+142)
196-196/? A/DEBUG:     #05 pc 00085053  /data/app/com.example-1/lib/arm/libtess.so (tesseract::Tesseract::recog_all_words(PAGE_RES*, ETEXT_DESC*, TBOX const*, char const*, int)+224)
196-196/? A/DEBUG:     #06 pc 0007b163  /data/app/com.example-1/lib/arm/libtess.so (tesseract::TessBaseAPI::Recognize(ETEXT_DESC*)+450)
196-196/? A/DEBUG:     #07 pc 0007b821  /data/app/com.example-1/lib/arm/libtess.so (tesseract::TessBaseAPI::GetUTF8Text(ETEXT_DESC*)+16)
196-196/? A/DEBUG:     #08 pc 001335bd  /data/app/com.example-1/lib/arm/libtess.so (Java_com_googlecode_tesseract_android_TessBaseAPI_nativeGetUTF8Text+76)
196-196/? A/DEBUG:     #09 pc 00ae3f11  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (java.lang.String com.googlecode.tesseract.android.TessBaseAPI.nativeGetUTF8Text()+76)
196-196/? A/DEBUG:     #10 pc 00ae6b61  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (java.lang.String com.googlecode.tesseract.android.TessBaseAPI.getUTF8Text()+348)
196-196/? A/DEBUG:     #11 pc 00bd17fb  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (com.example.ocr.OcrResultString com.example.ocr.Tesseract.startOcr(com.googlecode.leptonica.android.Pix, com.example.ocr.Options, com.example.ocr.Tesseract$OnOcrProgressListener, android.os.AsyncTask)+1038)
```

Processed with `ndk-stack`:
```
Build fingerprint: 'google/hammerhead/hammerhead:6.0.1/MMB29S/2489379:user/release-keys'
pid: 28285, tid: 28473, name: AsyncTask #1  >>> com.example <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x14
Stack frame 196-196/? A/DEBUG:     #00 pc 000782c8  /data/app/com.example-1/lib/arm/libtess.so (ELIST_ITERATOR::set_to_list(ELIST*)+1): Routine tesseract::TessBaseAPI::GetTextDirection(int*, float*) at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/api/baseapi.cpp:1957
Stack frame 196-196/? A/DEBUG:     #01 pc 000b8d27  /data/app/com.example-1/lib/arm/libtess.so (WERD::true_bounding_box() const+26): Routine tesseract::Tesseract::word_contains_non_1_digit(char const*, char const*) at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccmain/reject.cpp:520
Stack frame 196-196/? A/DEBUG:     #02 pc 000b8d69  /data/app/com.example-1/lib/arm/libtess.so (WERD::restricted_bounding_box(bool, bool) const+16): Routine tesseract::Tesseract::dont_allow_1Il(WERD_RES*) at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccmain/reject.cpp:530
Stack frame 196-196/? A/DEBUG:     #03 pc 000b8dd9  /data/app/com.example-1/lib/arm/libtess.so (WERD::bounding_box() const+8): Routine tesseract::Tesseract::dont_allow_1Il(WERD_RES*) at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccmain/reject.cpp:534
Stack frame 196-196/? A/DEBUG:     #04 pc 00084aab  /data/app/com.example-1/lib/arm/libtess.so (tesseract::Tesseract::RecogAllWordsPassN(int, ETEXT_DESC*, PAGE_RES_IT*, GenericVector<tesseract::WordData>*)+142): Routine ELIST2_ITERATOR::set_to_list(ELIST2*) at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccutil/elst2.h:279
Stack frame 196-196/? A/DEBUG:     #05 pc 00085053  /data/app/com.example-1/lib/arm/libtess.so (tesseract::Tesseract::recog_all_words(PAGE_RES*, ETEXT_DESC*, TBOX const*, char const*, int)+224): Routine ELIST at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccutil/elst.h:125
Stack frame 196-196/? A/DEBUG:     #06 pc 0007b163  /data/app/com.example-1/lib/arm/libtess.so (tesseract::TessBaseAPI::Recognize(ETEXT_DESC*)+450): Routine TessHOcrRendererCreate at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/api/capi.cpp:42
Stack frame 196-196/? A/DEBUG:     #07 pc 0007b821  /data/app/com.example-1/lib/arm/libtess.so (tesseract::TessBaseAPI::GetUTF8Text(ETEXT_DESC*)+16): Routine TessBaseAPIGetVariableAsString at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/api/capi.cpp:218
Stack frame 196-196/? A/DEBUG:     #08 pc 001335bd  /data/app/com.example-1/lib/arm/libtess.so (Java_com_googlecode_tesseract_android_TessBaseAPI_nativeGetUTF8Text+76): Routine IntCastRounded at \tess-two\tess-two/jni/com_googlecode_tesseract_android/src/ccutil/helpers.h:173
Stack frame 196-196/? A/DEBUG:     #09 pc 00ae3f11  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (java.lang.String com.googlecode.tesseract.android.TessBaseAPI.nativeGetUTF8Text()+76)
Stack frame 196-196/? A/DEBUG:     #10 pc 00ae6b61  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (java.lang.String com.googlecode.tesseract.android.TessBaseAPI.getUTF8Text()+348)
Stack frame 196-196/? A/DEBUG:     #11 pc 00bd17fb  /data/app/com.example-1/oat/arm/base.odex (offset 0x745000) (com.example.ocr.OcrResultString com.example.ocr.Tesseract.startOcr(com.googlecode.leptonica.android.Pix, com.example.ocr.Options, com.example.ocr.Tesseract$OnOcrProgressListener, android.os.AsyncTask)+1038)
```

**Comment by rmtheis** [comment]
Possibly fixed by #186. Needs testing.
